### DISCUSSION TITLE
token verification service on keycloak without client secret

### DISCUSSION THREAD

**Discussion by zoroglu** [keycloak/keycloak#25867] [Q&A]
### Description

When using keycloak **19.0.0**, I was trying to verify tokens without client_secret, but when switched to keycloak **23.0.3** version, I have to enter client_secret. Why did such a feature come?
If I don't turn on Client authentication from the Client settings. I'm getting a 403 Client not allowed error

**Keycloak 19.0.0**

Request Method: POST

URL: http://keycloak_url/realms/myrealm/protocol/openid-connect/token/introspect

BODY:

    token:{token}
    client_id:{client_id}

**Keycloak 23.0.3**

Request Method:POST

URL: http://keycloak_url/realms/myrealm/protocol/openid-connect/token/introspect

BODY:

    token:{token}
    client_id:{client_id}
    client_secret:{secret_id}

**Comment by shravankalee** [comment]
Hi, same issue for me.

I want to validate the token form the introspection endpoint but it is asking for client secret.
My client authentication is turned off, and I want to keep it that way. 
Is there any other way to validate a token with client authentication turned off?

**Comment by javagrails** [comment]
I didn't found any way to do that yet

**Comment by nicolaric** [comment]
any updates on this? i have the same issue...
